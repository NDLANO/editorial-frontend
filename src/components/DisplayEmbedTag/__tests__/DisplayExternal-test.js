/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import nock from 'nock';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import {
  DisplayExternal,
  getIframeSrcFromHtmlString,
} from '../DisplayExternal';

test('getIframeSrcFromHtmlString returns src attribute', () => {
  const src = getIframeSrcFromHtmlString(
    `<div>
      <iframe
        width="800"
        height="600"
        src="https://h5p-test.ndla.no/resource/3ab6850d">
      </iframe>
      <script src="https://h5p.org/h5p-resizer.js"></script>
    </div>`,
  );
  expect(src).toBe('https://h5p-test.ndla.no/resource/3ab6850d');
});

test('getIframeSrcFromHtmlString returns null id src not found', () => {
  const src = getIframeSrcFromHtmlString(
    `<div>
      <iframe
        width="800"
        height="600"
      </iframe>
      <script src="https://h5p.org/h5p-resizer.js"></script>
    </div>`,
  );
  expect(src).toBe(null);
});

test('DisplayOembed renderers correctly', () => {
  nock('http://ndla-api/')
    .persist()
    .get('/oembed-proxy/v1/oembed?url=https%3A%2F%2Fndla.no%2Foembed')
    .reply(200, {
      title: 'unit test',
      type: 'rich',
      html: '<iframe src="iframe.html">',
    });

  const component = TestRenderer.create(
    <DisplayExternal url="https://ndla.no/oembed" t={() => ''} />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('DisplayOembed display error on fetch fail', () => {
  nock('http://ndla-api/')
    .persist()
    .get('/oembed-proxy/v1/oembed?url=https%3A%2F%2Fndla.no%2Foembed')
    .replyWithError({
      message: 'something awful happened',
      code: 'AWFUL_ERROR',
    });

  const component = TestRenderer.create(
    <DisplayExternal url="https://ndla.no/oembed" t={() => 'Error message'} />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});
