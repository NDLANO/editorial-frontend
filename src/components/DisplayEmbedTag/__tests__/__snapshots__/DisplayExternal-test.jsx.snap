// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayOembed display error on fetch fail 1`] = `
Array [
  <button
    className="c-button c-button--stripped c-editor__delete-button"
    disabled={false}
    onClick={undefined}
    type="button"
>
    <svg
        className="c-icon"
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
            Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
        }
        viewBox="0 0 24 24"
        width="1em"
    >
        <title>
            Cross
        </title>
        <g>
            <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
                d="M0 0h24v24H0z"
                fill="none"
            />
        </g>
    </svg>
</button>,
  <div
    className="c-editor__error"
>
    <span>
        Error message
    </span>
</div>,
]
`;

exports[`DisplayOembed renderers correctly 1`] = `
Array [
  <div
    className="c-editor__figure-buttons"
>
    <button
        className="c-button c-button--stripped c-editor__button c-editor__button--red"
        disabled={false}
        onClick={undefined}
        type="button"
    >
        <svg
            className="c-icon"
            data-license="Apache License 2.0"
            data-source="Material Design"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
                Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
            }
            viewBox="0 0 24 24"
            width="1em"
        >
            <title>
                Cross
            </title>
            <g>
                <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
                <path
                    d="M0 0h24v24H0z"
                    fill="none"
                />
            </g>
        </svg>
    </button>
</div>,
  <div
    className="c-editor__error"
>
    <span>
        
    </span>
</div>,
]
`;

exports[`DisplayOembed renderers correctly 2`] = `
Array [
  <div
    className="c-editor__figure-buttons"
>
    <button
        className="c-button c-button--stripped c-editor__button c-editor__button--red"
        disabled={false}
        onClick={undefined}
        type="button"
    >
        <svg
            className="c-icon"
            data-license="Apache License 2.0"
            data-source="Material Design"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
                Object {
                    "color": undefined,
                    "verticalAlign": "middle",
                  }
            }
            viewBox="0 0 24 24"
            width="1em"
        >
            <title>
                Cross
            </title>
            <g>
                <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
                <path
                    d="M0 0h24v24H0z"
                    fill="none"
                />
            </g>
        </svg>
    </button>
</div>,
  <div
    className="c-editor__error"
>
    <span>
        
    </span>
</div>,
]
`;
